github/downloader_test.go

```diff
@@ -70,6 +61,44 @@ func loadOnlineTests(filepath string) (OnlineTests, error) {
 	return tests, nil
 }
 
+// checks whether a token exists as an env var, if not it skips the test
+func checkToken(t *testing.T) {
+	if os.Getenv("GITHUB_TOKEN") == "" {
+		t.Skip("GITHUB_TOKEN is not set")
+		return
+	}
+}
+
+func checkOS(t *testing.T) {
+	// docker service is not supported on osx in Travis: https://docs.travis-ci.com/user/docker/
+	if runtime.GOOS == "darwin" {
+		t.Skip("cannot run these tests on osx")
+		return
+	}
+}
+
+// Testing connection documentation, docker-compose and Migrate method
+func getDB(t *testing.T) (db *sql.DB) {
+	const DBURL = "postgres://user:password@127.0.0.1:5432/ghsync?sslmode=disable"
+	db, err := sql.Open("postgres", DBURL)
+	require.NoError(t, err, "DB URL is not working")
+	if err = db.Ping(); err != nil {
+		require.Nil(t, err, "DB connection is not working")
+	}
```

Comment:
there's a chance that the time gap between docker container run and this method is too small and, for example, port `5432` is still not available
we can probably add some exponential back off here https://github.com/cenkalti/backoff

Reply:
Thanks for the tip